Problem2.java

```diff
@@ -0,0 +1,86 @@
+//PROBLEM - Implement Hashset
+/** 3 Pointer Approcach: 
+ * 2d Arrrsy Boolean array.
+ * 1D array (Bucket) will be indexed [1, 1K] for Val%1000
+ * Each bucketItem array will be indexed for [1, 1K] from Val/1000
+ * We wiil initialize bucketItem arrays on the fly when needed and not in start as they may not be used.
+ * As it is given the numbers are in bound [1, 1M] so 1000 * 1000 will cover every number with NO collisions.
+ * Why Boolean array? - beacause we just need to check if specific value is available in the HashSet boolean True means value present else not.
+ * Approach - On Add item
+ * key%1000 - will give the bucket-Index; key/1000 - will give the bucketItem-Index;
+ */ 
+//
+//time Complexity :
+// add - Best/Amotized/Worst:O(1)
+// remove - Best/Amotized:O(1);
+// contains - Best/Amotized:O(1);
+
+
+// Space Complexity :
+// O(Buckets * BucketItems).
```

Comment:
Since **Buckets** and **BucketItems** are both 1000 in this case. The TC will be O(1), because O(1Million) is considered a constant.